src/core/hooks/usePromiseCache.ts

```diff
@@ -26,5 +21,8 @@ export default function usePromiseCache(
         delete promises[cacheKey];
       });
     },
+    getExisting() {
+      return promises[cacheKey];
+    },
```

Comment:
I think the name of this function could be `getExistingPromise()` instead, because that's what I would like to call it when it's used, and we might as well give it a good default name.